### PR TITLE
🔨 Maintenance/pyupgrade code base targeting py3.9

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

The entire repo (*) use python 3.9. 
Upgrading to the new syntax (e.g. ``List`` -> ``list``-like ) is very time consuming, noisy and error-prone. 

This PR uses [pyupgrade](https://github.com/asottile/pyupgrade) to automatically upgrade targeted code syntax to py3.9
- 🔥 we start (this PR) with some non-critical code base in the repo (namely ``tests``, ``api`` and ``packages/settings-library``). Subsequent PRs  will address packages and then one service at a time.
- 🔨 Adds pre-commit hook that applies pyupgrade to modified code
- 🔨Adds ``scripts/pyupgrade.bash`` 
- 📝 adds some doc on python versioning (draft) since this tool can also assist in future python's upgrades

(*) Except for ``services/director`` and some files in ``packages/pytest-simcore`` used as well there

## How to test

- 100% test coverage should ideally cover this

## TODO
- [x] pre-commit hook
- [x] shell script in ``scripts/`` with containarized tool
- [x] make recipe
- [x] document py ugrade

### CODE REVIEW

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 1]
```diff
@@ -255,38 +250,26 @@ def docker_stack(
...
-
-            asyncio.run(_check_all_services_are_running())
+        from tenacity._asyncio import AsyncRetrying
 
-        else:
-            for attempt in Retrying(
-                wait=wait_fixed(5),
+        async def _check_all_services_are_running():
+            async for attempt in AsyncRetrying(
+                wait=wait_fixed(1),
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:257]: I wonder why we had 1 second with the AsyncRetrying and 5 seconds with the Retrying

File: packages/pytest-simcore/src/pytest_simcore/helpers/utils_tokens.py [hunk 2]
```diff
@@ -50,7 +49,11 @@ async def get_token_from_db(
             token_service=token_service,
             token_data=token_data,
         )
-        stmt = sa.select([tokens,]).where(expr)
+        stmt = sa.select(
+            [
+                tokens,
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/helpers/utils_tokens.py:54]: remove coma and reformat
```suggestion
                tokens
```

File: tests/performance/locust_files/platform_ping_test.py [hunk 3]
```diff
@@ -6,14 +6,18 @@
...
 from locust import task
 from locust.contrib.fasthttp import FastHttpUser
 
 logging.basicConfig(level=logging.INFO)
 
 
+# 'import locust_plugins' is necessary to use --check-fail-ratio
+assert locust_plugins  # nosec
+# adds assert here to avoid pycln removing it
+
```
Comment (GitHK) [inline tests/performance/locust_files/platform_ping_test.py:20]: not sure I fully understand why adding the assert is requried

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 4]
```diff
@@ -255,38 +250,26 @@ def docker_stack(
...
                 before_sleep=before_sleep_log(log, logging.INFO),
                 reraise=True,
             ):
                 with attempt:
-                    for service in docker_client.services.list():
-                        assert_service_is_running(service)
+                    await asyncio.gather(
+                        *[
+                            asyncio.get_event_loop().run_in_executor(
+                                None, assert_service_is_running, service
```
Comment (GitHK) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:266]: Now that I look at this `assert_service_is_running` is blocking or is it just me? It's not even a coroutine.

**Review by GitHK — APPROVED** [review]
Nice! Glad this will go out of the way.
Please see below some minor stuff and questions.

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py (hunk 1 above)
Comment (pcrespov) [reply packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:257]: good catch! I did a c&p from an old version ...

File: tests/performance/locust_files/platform_ping_test.py (hunk 3 above)
Comment (pcrespov) [reply tests/performance/locust_files/platform_ping_test.py:20]: improved doc

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py (hunk 4 above)
Comment (pcrespov) [reply packages/pytest-simcore/src/pytest_simcore/docker_swarm.py:266]: It is to parallelize check. It does not need to be a coroutine to run in the executor ...  SEE https://docs.python.org/3/library/asyncio-eventloop.html#executing-code-in-thread-or-process-pools

**Comment by pcrespov** [comment]
> 

Yes. Easy. That is why this PR only upgrades non-critical packages
In addition, there is now the pre-commit hook. So from now on, whatever gets changed it is also upgraded on the fly

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -11,7 +11,13 @@ repos:
       - id: check-yaml
       - id: check-added-large-files
       - id: check-merge-conflict
-  # NOTE: Keep order as pycln (remove unused imports), then isort (sort them) and black (final formatting)
+  # NOTE: Keep order as pyupgrade (will update code) then pycln (remove unused imports), then isort (sort them) and black (final formatting)
```
Comment (dyollb) [inline .pre-commit-config.yaml:14]: if you pass `args: ["--profile", "black"]` to isort it won't conflict with black

**Comment by dyollb** [comment]
Nice!

**Pull request merged**